tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+
+        self.assertEqual(
+            len(
+                [
+                    port_group
+                    for nested_group in [
+                        port_group.portfolios
+                        for port_group in responses["portfolio_groups"]["success"]
+                    ]
+                    for port_group in nested_group
+                ]
+            ),
+            len(data_frame),
+        )
+
+    def test_one_port_groups_no_portfolios(self) -> None:
+
+        """
+        Description:
+        ------------
+        Here we test adding one new portfolio group with no portfolios.
+        """
+
+        data_frame = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_group_no_portfolio.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
+
+        self.assertEqual(
+            first=len(
+                [
+                    port_group._id
+                    for port_group in responses["portfolio_groups"]["success"]
+                ]
+            ),
+            second=len(data_frame),
+        )
+
+        self.assertEqual(
+            first=responses["portfolio_groups"]["success"][0]._id,
+            second=lusid.models.ResourceId(scope=self.scope, code="PG_TEST3A"),
+        )
+
+    def test_multiple_port_groups_no_portfolios(self) -> None:
+
+        """
+        Description:
+        ------------
+        Here we test adding multiple new portfolio group with no portfolios.
+        """
+
+        data_frame = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_groups_no_portfolio.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
```

Comment:
Can you log this instead of printing it please?

Reply:
Nice work, in log_error_requests_title please use an if statement rather than try except.